ingestion/ingestion_server/ingestion_file_upload_rest_api.py

```diff
@@ -159,82 +159,101 @@ def file_upload():
+                                                                               creator_domain="www.magen.io",
+                                                                               iv=iv_decoded,
+                                                                               enc_asset_hash=b64_file_digest.hexdigest())
+                    metadata_b64 = ContainerApi.b64encode_meta_v2(metadata_json)
+                    metadata_b64_str = metadata_b64.decode("utf-8")
+                    html_container_path = dst_file_path + ".html"
+
+                    if not ContainerApi.create_html_file_container_from_file(metadata_dict, metadata_b64_str,
+                                                                             base64_file_path, html_container_path):
+                        raise Exception("Failed to create container: {}".format(dst_file_path))
+                else:
+                    raise Exception("Key Server problem")
+
+            # GridFS file storage
+            db_core = MainDb.get_core_db_instance()
+            # TODO bucket name, owner and group should be based on user login such as email
+            if ext == 'pub':
+                html_container_path = dst_file_path + ".html"
                 metadata_json, metadata_dict = ContainerApi.create_meta_v2(asset_dict,
-                                                                           creator_domain="www.magen.io",
-                                                                           iv=iv_decoded,
-                                                                           enc_asset_hash=b64_file_digest.hexdigest())
+                                                                           creator_domain="www.magen.io")
                 metadata_b64 = ContainerApi.b64encode_meta_v2(metadata_json)
                 metadata_b64_str = metadata_b64.decode("utf-8")
-                html_container_path = dst_file_path + ".html"
-                if not ContainerApi.create_html_file_container_from_file(metadata_dict, metadata_b64_str,
-                                                                         base64_file_path, html_container_path):
+                if not ContainerApi.create_html_file_container_from_file(metadata_dict, metadata_b64_str, None, html_container_path):
                     raise Exception("Failed to create container: {}".format(dst_file_path))
 
-                # GridFS file storage
-                db_core = MainDb.get_core_db_instance()
-                # TODO bucket name, owner and group should be based on user login such as email
-                with open(html_container_path, "rb") as magen_file_upload:
-                    fs = gridfs.GridFSBucket(db_core.get_magen_mdb())
+            with open(html_container_path, "rb") as magen_file_upload:
+                fs = gridfs.GridFSBucket(db_core.get_magen_mdb())
+                if ext == 'pub':
```

Comment:
We are again testing if ext == pub. the code here needs to be rearchitected to avoid such high number of branches.